src/D2L.CodeStyle.Annotations/Contract/OnlyVisibleToAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace D2L.CodeStyle.Annotations.Contract {
+
+	[AttributeUsage(
+		validOn: AttributeTargets.Method | AttributeTargets.Property,
+		AllowMultiple = true,
+		Inherited = false

```

Comment:
Not going to support inheritance for the purposes of speed and readability.  If we want, we can always add the ability to force it down the inheritance chain, but require it everywhere like `[Immutable]`.  That way if you go to the definition, you always see the list.